aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/PersistentMutationOutbox.java

```diff
@@ -414,5 +425,25 @@ private Completable unexpectedMutationScenario() {
                 "Please report at https://github.com/aws-amplify/amplify-android/issues."
             ));
         }
+
+        private PendingMutation<Model> merge(PendingMutation<T> incoming, PendingMutation<T> existing)
+                throws AmplifyException {
+            SerializedModel incomingSerializedModel = SerializedModel.create(incoming.getMutatedItem(),
+                                                                            incoming.getModelSchema());
+            SerializedModel existingSerializedModel = SerializedModel.create(existing.getMutatedItem(),
+                                                                            existing.getModelSchema());
+            Map<String, Object> mergedSerializedData = new HashMap<>();
+            for (String key : incomingSerializedModel.getSerializedData().keySet()) {
+                Object value = incomingSerializedModel.getSerializedData().get(key) != null ?
```

Comment:
If the value is null, that is actually a valid value that the customer may want to set this field to.  What we should be checking here instead, is if the key _exists_.